diff -up nano-8.1/config.h.in.omv~ nano-8.1/config.h.in
--- nano-8.1/config.h.in.omv~	2024-07-12 21:23:02.673981833 +0200
+++ nano-8.1/config.h.in	2024-07-12 21:23:26.062250504 +0200
@@ -2149,7 +2149,7 @@
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if _GL_GNUC_PREREQ (11, 0)
+# if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up nano-8.1/m4/gnulib-common.m4.omv~ nano-8.1/m4/gnulib-common.m4
--- nano-8.1/m4/gnulib-common.m4.omv~	2024-07-12 21:25:29.235664279 +0200
+++ nano-8.1/m4/gnulib-common.m4	2024-07-12 21:25:37.639760677 +0200
@@ -223,7 +223,7 @@ AC_DEFUN([gl_COMMON_BODY], [
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if _GL_GNUC_PREREQ (11, 0)
+# if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
